editor/scene/sprite_frames_editor_plugin.cpp

```diff
@@ -324,20 +324,49 @@ void SpriteFramesEditor::_sheet_zoom_on_position(float p_zoom, const Vector2 &p_
 }
 
 void SpriteFramesEditor::_sheet_zoom_in() {
+	sheet_zoom_fitted = false;
 	_sheet_zoom_on_position(scale_ratio, Vector2());
 }
 
 void SpriteFramesEditor::_sheet_zoom_out() {
+	sheet_zoom_fitted = false;
 	_sheet_zoom_on_position(1 / scale_ratio, Vector2());
 }
 
 void SpriteFramesEditor::_sheet_zoom_reset() {
+	sheet_zoom_fitted = false;
 	// Default the zoom to match the editor scale, but don't dezoom on editor scales below 100% to prevent pixel art from looking bad.
 	sheet_zoom = MAX(1.0f, EDSCALE);
 	Size2 texture_size = split_sheet_preview->get_texture()->get_size();
 	split_sheet_preview->set_custom_minimum_size(texture_size * sheet_zoom);
 }
 
+void SpriteFramesEditor::_sheet_zoom_fit() {
+	sheet_zoom_fitted = true;
+
+	const float margin_percentage = 0.1f;
+	const Size2 max_margin = Vector2(64.0f, 64.0f);
+	const Size2 margin = (margin_percentage * split_sheet_scroll->get_size()).min(max_margin);
+	const Size2 display_area_size = split_sheet_scroll->get_size() - margin;
+	const Size2 texture_size = split_sheet_preview->get_texture()->get_size();
+	const Vector2 texture_ratio = display_area_size / texture_size;
+	const float texture_fit_zoom = MIN(texture_ratio.x, texture_ratio.y);
+
+	const Size2 min_frame_size = Size2(64.0, 64.0);
+	const Size2 frame_size = _get_frame_size().max(Size2i(1, 1));
```

Comment:
```suggestion
	const Size2 frame_size = _get_frame_size().maxi(1);
```
Unless it is a `Size2`, then it should be
```suggestion
	const Size2 frame_size = _get_frame_size().maxf(1.0);
```